packages/core/admin/ee/server/services/review-workflows/stages.js

```diff
@@ -1,36 +1,88 @@
 'use strict';
 
+const { mapAsync } = require('@strapi/utils');
+
 const { STAGE_MODEL_UID } = require('../../constants/workflows');
+const { getService } = require('../../utils');
+
+module.exports = ({ strapi }) => {
+  const workflowsService = getService('workflows', { strapi });
+
+  return {
+    find({ workflowId, populate }) {
+      const params = {
+        filters: { workflow: workflowId },
+        populate,
+      };
+      return strapi.entityService.findMany(STAGE_MODEL_UID, params);
+    },
+
+    findById(id, { workflowId, populate }) {
+      const params = {
+        filters: { workflow: workflowId },
+        populate,
+      };
+      return strapi.entityService.findOne(STAGE_MODEL_UID, id, params);
+    },
+
+    createMany(stagesList, { fields }) {
+      const params = {
+        select: fields,
+      };
+      return Promise.all(
+        stagesList.map((stage) =>
+          strapi.entityService.create(STAGE_MODEL_UID, { data: stage, ...params })
+        )
+      );
+    },
+
+    update(stageId, stageData) {
+      return strapi.entityService.update(STAGE_MODEL_UID, stageId, { data: stageData });
+    },
+
+    delete(stageId) {
+      return strapi.entityService.delete(STAGE_MODEL_UID, stageId);
+    },
+
+    count() {
+      return strapi.entityService.count(STAGE_MODEL_UID);
+    },
+
+    async replaceWorkflowStages(workflowId, stages) {
+      const workflow = await workflowsService.findById(workflowId, { populate: ['stages'] });
+
+      const { created, updated, deleted } = getDiffBetweenStages(workflow.stages, stages);
+
+      const newStages = await this.createMany(created, { fields: ['id'] });
+      const assignedStagesToWorkflow = stages.map((stage) => stage.id ?? newStages.shift().id);
```

Comment:
minor suggestion, why if we call this variable `stagesIds` ? Adding the word assigned here feels a bit out of place.
```suggestion
      const stagesIds = stages.map((stage) => stage.id ?? newStages.shift().id);
```

Reply:
I thought of that. But I think that when reading, it would be easy to mistake `stagesIds` as being the ids in the `stages` list when they are not.

Reply:
But aren't they the ids of the `stages` ? The ids that are not new are filled , but they still are the stage ids no?

Reply:
I guess you're right. 🤔